### PR TITLE
Deprecate DELETE/GET regions/:id

### PR DESCRIPTION
**Pull request by mhoppa**
## What does this PR (Pull Request) do?

- [x] This PR is not related to any Issue 

## Which Traffic Control components are affected by this PR?

- Documentation
- Traffic Ops

## What is the best way to verify this PR?

Run traffic ops and make sure you can not hit the endpoint in API 2.0 and when performing a GET on it in 1.x you receive a deprecation message back in the returned alerts. 

## The following criteria are ALL met by this PR

- [x] This PR includes tests OR I have explained why tests are unnecessary
- [x] This PR includes documentation OR I have explained why documentation is unnecessary
- [x] This PR includes an update to CHANGELOG.md OR such an update is not necessary
- [x] This PR includes any and all required license headers
- [x] This PR ensures that database migration sequence is correct OR this PR does not include a database migration
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://www.apache.org/security/) for details)

### CODE REVIEW

File: traffic_ops/traffic_ops_golang/api/api.go [hunk 1]
```diff
@@ -158,7 +158,6 @@ func HandleDeprecatedErr(w http.ResponseWriter, r *http.Request, tx *sql.Tx, sta
 		log.Errorf("HandleDeprecatedErr called after a write already occurred! Attempting to write the error anyway! Path %s", r.URL.Path)
 		// Don't return, attempt to rollback and write the error anyway
 	}
-	setRespWritten(r)
```
[the remark this answers is not in the document]
Comment (ocket8888) [reply traffic_ops/traffic_ops_golang/api/api.go]: Just need an answer to this, I assume it's being done somewhere twice that I'm missing.

[the remark this answers is not in the document]
Comment (mhoppa) [reply traffic_ops/traffic_ops_golang/api/api.go]: yeah its done in the `WriteAlerts` function as well which was causing issues

**Comment by ocket8888** [comment]
Now just gotta fix those conflicts

**Comment by mhoppa** [comment]
done!

**Pull request merged**